### DISCUSSION TITLE
How to prevent github pages deployment builds from being count into the contribution graph?

### DISCUSSION THREAD

**Discussion by kkew3** [community/community#147805] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

So here is my contribution graph:

![contribution_graph](https://github.com/user-attachments/assets/e17f5123-02d8-43ed-8546-8cbb76f66971)

But the deployment commits of my github pages submitted by bots take up the majority (see "kkew3/kkew3.github.io 522 commits"; I never wrote so many posts!). Therefore, it does not reflect my real contribution at all. Is there anyway to fix this? Thank you so much!

**Comment by mecodeatlas** [comment, marked as answer]
Hey @kkew3, thanks for reaching out!

In one of our forums, I found [this discussion](https://github.com/actions/checkout/discussions/479) where another user faced an issue similar to what you're experiencing. This might be a potential solution for you.

Additionally, if you want to prevent deployment commits from being attributed to your profile without changing the configuration for the entire repository, you can update your deployment workflow to attribute those commits to the GitHub Actions bot. Here’s an example of how to do that:

`git -c user.name="github-actions[bot]" -c user.email="github-actions[bot]@users.noreply.github.com" commit -m "GitHub Actions Commit"`

This approach ensures that the change applies only to deployment commits and won’t override your global Git configuration.

Hope it helps :)

**Reply by kkew3** [reply]
Thank you so much! Since I'm using [`jeffreytse/jekyll-deploy-action`](https://github.com/jeffreytse/jekyll-deploy-action), I asked the author's advice on the change suggested here. The working example, according to the author, is:

```diff
   - uses: jeffreytse/jekyll-deploy-action@v0.6.0
      with:
        provider: 'github'
        token: ${{ secrets.GITHUB_TOKEN }} # It's your Personal Access Token(PAT)
        repository: ''             # Default is current repository
        branch: 'gh-pages'         # Default is gh-pages for github provider
        jekyll_src: './'           # Default is root directory
        jekyll_cfg: '_config.yml'  # Default is _config.yml
        jekyll_baseurl: ''         # Default is according to _config.yml
        bundler_ver: ''            # Default is compatible bundler version (~>2.5.0)
        ruby_ver: '3.1.3'
        cname: ''                  # Default is to not use a cname
-       actor: ''                  # Default is the GITHUB_ACTOR
+       actor: 'github-actions[bot]' # Default is the GITHUB_ACTOR
        pre_build_commands: |
          # Installing additional dependencies (Arch Linux)
          git config --global http.version HTTP/1.1
          git config --global http.postBuffer 157286400
```
